pkg/ui/console/client.go

```diff
@@ -338,13 +338,19 @@ func (c Client) UpgradeProviders(ctx context.Context, args []string) error {
 		return err
 	}
 	ui.ColorizedOutput(ui.ColorProgress, "Upgrading CloudQuery providers %s\n\n", args)
+	dupes := make(map[string]struct{}, len(providers))
```

Comment:
dupes shouldn't be allowed, as it won't parse in the config.

Reply:
It works with aliased things... and upgrades aws 2 times

Reply:
I see what this fix does now, actually would make a function on the provider list to return a unique set and then iterate over that, or actually for upgrade it should iterate on the top level providers and not the provider blocks.

Reply:
for 0.21 I assume.

Reply:
Open an issue for this?